File: ios/BlueViewWeather/Views/Components/ForecastRow.swift
import SwiftUI

struct ForecastRow: View {
    let daily: DailyDto
    let units: String
    let selectedDate: String?
    let onDayClick: (String) -> Void

    private var sym: String { units == "imperial" ? "°F" : "°C" }
    private var today: String { isoDateString(Date()) }

    var body: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: 8) {
                ForEach(daily.time.indices, id: \.self) { i in
                    let date = daily.time[i]
                    let selected = date == selectedDate
                    let dayLabel = date == today ? "Today" : shortWeekday(from: date)

                    DayCard(
                        day: dayLabel,
                        icon: wmoIcon(daily.weatherCode[i], isDay: true),
                        hi: "\(Int(daily.tempMax[i]))\(sym)",
                        lo: "\(Int(daily.tempMin[i]))\(sym)",
                        pop: daily.precipProbMax.indices.contains(i) ? (daily.precipProbMax[i] ?? 0) : 0,
                        selected: selected,
                        onClick: { onDayClick(date) }
                    )
                }
            }
            .padding(.horizontal, 12)
            .padding(.vertical, 10)
        }
    }
}

private struct DayCard: View {
    let day: String
    let icon: String
    let hi: String
    let lo: String
    let pop: Int
    let selected: Bool
    let onClick: () -> Void

    var body: some View {
        Button(action: onClick) {
            VStack(spacing: 3) {
                Text(day)
                    .font(.caption2)
                    .foregroundColor(.textSecondary)
                    .lineLimit(1)
                Text(icon)
                    .font(.system(size: 26))
                Text(hi)
                    .font(.subheadline.weight(.bold))
                    .foregroundColor(.tempHigh)
                Text(lo)
                    .font(.subheadline)
                    .foregroundColor(.tempLow)
                if pop >= 10 {
                    Text("💧\(pop)%")
                        .font(.caption2)
                        .foregroundColor(.rainBlue)
                }
            }
            .frame(width: 60)
            .padding(.vertical, 10)
            .padding(.horizontal, 4)
            .background(selected ? Color.blueAccent.opacity(0.15) : Color.navyCard)
            .overlay(
                RoundedRectangle(cornerRadius: 10)
                    .stroke(selected ? Color.blueAccent.opacity(0.5) : Color.navyBorder, lineWidth: 1)
            )
            .clipShape(RoundedRectangle(cornerRadius: 10))
        }
        .buttonStyle(.plain)
    }
}

func isoDateString(_ date: Date) -> String {
    let formatter = DateFormatter()
    formatter.dateFormat = "yyyy-MM-dd"
    formatter.calendar = Calendar(identifier: .gregorian)
    formatter.timeZone = TimeZone.current
    return formatter.string(from: date)
}

func shortWeekday(from isoDate: String) -> String {
    let inFormatter = DateFormatter()
    inFormatter.dateFormat = "yyyy-MM-dd"
    guard let date = inFormatter.date(from: isoDate) else { return isoDate }
    let outFormatter = DateFormatter()
    outFormatter.dateFormat = "EEE"
    return outFormatter.string(from: date)
}

File: app/src/main/java/ai/blueview/weather/ui/components/ForecastRow.kt
package ai.blueview.weather.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import ai.blueview.weather.data.api.dto.DailyDto
import ai.blueview.weather.ui.theme.*
import ai.blueview.weather.util.wmoIcon
import java.time.LocalDate
import java.time.format.TextStyle as JTextStyle
import java.util.Locale

@Composable
fun ForecastRow(
    daily: DailyDto,
    units: String,
    selectedDate: String?,
    onDayClick: (String) -> Unit,
    modifier: Modifier = Modifier
) {
    val sym   = if (units == "imperial") "°F" else "°C"
    val today = LocalDate.now().toString()

    LazyRow(
        modifier            = modifier.fillMaxWidth(),
        contentPadding      = PaddingValues(horizontal = 12.dp, vertical = 10.dp),
        horizontalArrangement = Arrangement.spacedBy(8.dp)
    ) {
        items(daily.time.indices.toList()) { i ->
            val date     = daily.time[i]
            val selected = date == selectedDate
            val dayLabel = if (date == today) "Today"
            else LocalDate.parse(date).dayOfWeek
                .getDisplayName(JTextStyle.SHORT, Locale.getDefault())

            DayCard(
                day      = dayLabel,
                icon     = wmoIcon(daily.weatherCode[i], isDay = true),
                hi       = "${daily.tempMax[i].toInt()}$sym",
                lo       = "${daily.tempMin[i].toInt()}$sym",
                pop      = daily.precipProbMax.getOrNull(i) ?: 0,
                selected = selected,
                onClick  = { onDayClick(date) }
            )
        }
    }
}

@Composable
private fun DayCard(
    day: String, icon: String, hi: String, lo: String,
    pop: Int, selected: Boolean, onClick: () -> Unit
) {
    val shape  = RoundedCornerShape(10.dp)
    val bgColor = if (selected) BlueAccent.copy(alpha = 0.15f)
                  else NavyCard
    val border  = if (selected) BlueAccent.copy(alpha = 0.5f)
                  else NavyBorder

    Column(
        modifier = Modifier
            .width(60.dp)
            .clip(shape)
            .background(bgColor)
            .border(1.dp, border, shape)
            .clickable(onClick = onClick)
            .padding(vertical = 10.dp, horizontal = 4.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(3.dp)
    ) {
        Text(day,  style = MaterialTheme.typography.labelSmall, color = TextSecondary,
            textAlign = TextAlign.Center, maxLines = 1)
        Text(icon, fontSize = 26.sp, textAlign = TextAlign.Center)
        Text(hi,   style = MaterialTheme.typography.bodyMedium, color = TempHigh,
            fontWeight = FontWeight.Bold)
        Text(lo,   style = MaterialTheme.typography.bodyMedium, color = TempLow)
        if (pop >= 10)
            Text("💧$pop%", style = MaterialTheme.typography.labelSmall, color = RainBlue)
    }
}
